docs/revsets.md

```diff
@@ -70,6 +70,29 @@ only symbols.
 You can use parentheses to control evaluation order, such as `(x & y) | z` or
 `x & (y | z)`.
 
+### The `all:` prefix
+
+An `all:` prefix before a revset expression does not change its meansing.
+Strictly speaking, this is not part of the revset language.
+
+Some arguments to some commands expect a revset expression that expands to a
+single revision. The `all:` prefix overrides this restriction for many such
+arguments.
+
+For example, `jj rebase -r w -d xyz+` will rebase `w` on top of the child of
+`xyz` as long as `xyz` has exactly one child.
+
+If `xyz` has more than one child, without the `all:` prefix, `jj rebase -r w -d
+xyz+` will return an error.
+
+With the `all:` prefix and if if `xyz` has more than one child, `jj rebase -r w
+-d all:xyz+` will make `w` into a merge commit. The `all:` prefix confirms that
+the user expected `xyz` to have more than one child.
+
+A more useful example: if `w` is a merge commit, `jj rebase -s w -d all:w- -d
```

Comment:
TODO suggested by @bnjmnt4n on Discord:


> is it worth leaving a note about the new option from https://github.com/martinvonz/jj/pull/3458 as well?

The option is `ui.always-allow-large-revsets`

Reply:
PTAL. I moved most of the docs to `revsets.md`.

Reply:
The move LGTM!